docs: update AppKit terminology to Reown across documentation (#738)
reown-com/reown-docs: appkit/react-native/core/migration-multichain.mdx

This step is critical for avoiding runtime errors related to missing polyfills.

> ## 3. Reown Initialization (`createAppKit`)

Reviewer: ```suggestion
## 3. Reown AppKit Initialization (`createAppKit`)
```

> The way you initialize Reown has changed significantly to support the new adapter-based architecture.

Reviewer: ```suggestion
The method of initializing the SDK has changed to support adapter-based architecture.
```

-   `createAppKit` is imported from `@reown/appkit-react-native`.
-   You explicitly create instances of the adapters you need (e.g., `new EthersAdapter()`, `new WagmiAdapter(...)`).
-   The `adapters` array in `createAppKit` takes these initialized adapter instances.

> -   The `networks` array in `createAppKit` takes `AppKitNetwork` objects, which define all supported networks for Reown's UI and context. `viem/chains` objects can be directly used for EVM networks.

Reviewer: ```suggestion
-   The `networks` array in `createAppKit` takes `AppKitNetwork` objects, which define all supported networks for Reown AppKit's UI and context. `viem/chains` objects can be directly used for EVM networks.
```
